tracks/python/exercises/say/mentoring.md

```diff
@@ -0,0 +1,128 @@
+denominations = (9, "billion"), (6, "million"), (3, "thousand"), (2, "hundred")
+
+def say(number):
+    if 0 > number or number > 999999999999:
+        raise ValueError("input out of range")
+    if 10 <= number < 20:
+        return teens[number % 10]
+    if number < 100:
+        if number // 10 and number % 10:
+            return "-".join([tens[number // 10], digits[number % 10]])
+        else:
+            return tens[number // 10] or digits[number % 10] or "zero"
+    for k, denomination in denominations:
+        if number > 10**k - 1:
+            result = [say(number // 10**k), denomination]
+            if number % 10 ** k:
+                result.append(say(number % 10**k))
+            return " ".join(result)
+```
+Or without using recursion:
+```
+denominations = (9, "billion"), (6, "million"), (3, "thousand"), (0, "")
+
+def say_three_digits(number):
+    hundreds_col, tens_col, ones_col = number // 100, number // 10 % 10, number % 10
+    result = [str(digits[hundreds_col]) + " hundred"] if hundreds_col else []
+    if tens_col == 1:
+        result.append(teens[ones_col])
+    elif tens_col and ones_col:
+        result.append("-".join([tens[tens_col], digits[ones_col]]))
+    else:
+        result.append(tens[tens_col] or digits[ones_col])
+    return result
+
+def say(number):
+    if 0 > number or number > 999999999999:
+        raise ValueError("input out of range")
+    result = []
+    for k, denomination in denominations:
+        if number > 10**k - 1:
+            result += say_three_digits(number // 10**k)
+            result += [denomination]
+            number %= 10**k
+    return " ".join(e for e in result if e) or "zero"
+```
+
+## Common Suggestions
+
+### Defining the literals
+Typically this problem will begin with defining the literals needed. 
+These can be done as either lists or dictionaries; many students will find dictionaries easier to work with for this problem. 
+Encourage the student to define these at the start rather than embedding them in their code.
+
+### Values below 1000
+Next, the instructions suggest begining by creating a function which works for values of up to three digits.
+The return value of this function will typically be a string, but it can also be a list of strings for later assembly.
+Whether in this function or for the final return, encourage them to use a list and combine the elements using `str.join`.
+
+The main challenges are ensuring that the 'hundred' denominator is conditionally included and that the tens and ones columns are joined with a hyphen if they are both present.
+Simple `if` checks for each of these conditions is the most straightforward approach.
+
+Another approach is to only handle values below 100, and instead handle the hundreds alongside the other denominators.
+This typically requires the use of recursion. 
+
+### Larger numbers
+Encourage the student to use the first function they wrote for each set of three digits rather than duplicating code.
```

Comment:
```suggestion
### Larger numbers

Encourage the student to use the first function they wrote for each set of three digits rather than duplicating code.
```